common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RetryingBlockFetcher.java

```diff
@@ -178,6 +213,12 @@ private synchronized boolean shouldRetry(Throwable e) {
     boolean isIOException = e instanceof IOException
       || (e.getCause() != null && e.getCause() instanceof IOException);
     boolean hasRemainingRetries = retryCount < maxRetries;
+    if (isIOException && hasRemainingRetries && executorAliveChecker != null) {
+       if (!executorAliveChecker.check()) {
+         throw new ExecutorDeadException("The relative remote executor," +
```

Comment:
can we throw the exception earlier? like in `fetchStarter.createAndStart`.

Reply:
Yes, done.